crates/telio-wg/src/link_detection/mod.rs

```diff
@@ -111,7 +148,21 @@ impl LinkDetectionEnabled {
         }
 
         if let Some(state) = self.peers.get_mut(public_key) {
-            state.update(rx_bytes, tx_bytes, node_state, self.rtt)
+            let result = state.update(
+                rx_bytes,
+                tx_bytes,
+                node_state,
+                self.rtt,
+                self.enhanced_detection,
+            );
+
+            if self.enhanced_detection && result.should_ping {
+                if let Err(e) = self.ping_channel.send(node_addresses).await {
+                    telio_log_warn!("Failed to trigger ping to {:?} {:?}", public_key, e);
+                }
+            }
```

Comment:
Hmm.... What happens if pinging is in action but link detection noticed that the state is up because it received the packet? Does the state machine handle such a transition well? It's a bit hard to follow the code so I'm asking instead :)